core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -137,6 +137,12 @@
      */
     TableStatistics getTableStatistics(Session session, TableHandle tableHandle, Constraint constraint);
 
+    /**
+     * Returns {@code true} when {@link #getTableStatistics} returns more useful (e.g. more accurate) result
+     * for {@code oldTableHandle} than for {@code newTableHandle}.
+     */
+    boolean degradesTableStatisticsAccuracy(Session session, List<TableHandle> oldTableHandles, TableHandle newTableHandle);
```

Comment:
I'm not sure I understand the semantics of this method. What does it mean for a table handle to produce more or less accurate statistics? Statistics are either better or worse, but accuracy is a tricky thing to measure, and you'd need to know *what* you're providing statistics for.

Also, what happens if the caller provides a handle, say, for something that represents `SELECT * FROM t WHERE ...` and another for `SELECT * FROM u WHERE ....`. Which one is more "accurate"?

Reply:
> Also, what happens if the caller provides a handle, say, for something that represents `SELECT * FROM t WHERE ...` and another for `SELECT * FROM u WHERE ....`. Which one is more "accurate"?

That's partial ordering. Connector doesn't need to distinguish between unrelated table handles (and engine won't ever ask for these).
We should clarify this in documentation. can you suggest appropriate wording?

>  What does it mean for a table handle to produce more or less accurate statistics?

If some attributes of Table Handle would impact statistics results, but are ignored by the connector, then such statistics are "worse" or "less accurate".

For example, imagine hive connector always returned table statistics regardless of the filter.
Then such statistics would be appropriate for a table handle that doesn't have a filter,
but would be less accurate for a table handle that does have it.

Reply:
But how would it know what statistics the caller is asking about? Wouldn't the method have to, at a minimum, take a TableStatistics value?

Reply:
@martint the statistics that would be returned by `getTableStatistics` call, as javadoc aimed to suggest.